staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -128,10 +129,23 @@ func tlsConfigKey(c *Config) (tlsCacheKey, error) {
 	if err := loadTLSFiles(c); err != nil {
 		return tlsCacheKey{}, err
 	}
+
+	// since golang forbids comparing functions we should take the address of the certificate instead
```

Comment:
What if we were to change the entire interface to  require something with a name and we set the name to "dynamic:certfile:keyfile" or more generally, "scheme:something".  It would be more cumbersome and larger, but it would be very explicit.

Reply:
somehting like `TLSConfig.NewGetCert func() (*tls.Certificate, identity string, error)` ?